subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -390,6 +383,7 @@ private void parseData(String data) {
                 requestFlag += 1;
               } else {
                 System.out.println("finished running configuration requests");
+                validateTests();
                 System.exit(0);
```

Comment:
System.exit(0) is very bad practice in the middle of a function.  Like "yikes" bad -- especially on success. What this really means is that you shouldn't be running this (parseData) in a separate thread. Anyway -- likely not something you can easily clean up now -- and I should have caught this before (or maybe I did and my comment got lost) -- but taking a quick look at your code I think you're way over-using Threads...

Reply:
fixed with proper closing down of threads. system comes to a clean exit